airflow/providers/amazon/aws/executors/ecs/ecs_executor.py

```diff
@@ -493,3 +496,39 @@ def get_container(self, container_list):
                     'container "name" must be provided in "containerOverrides" configuration'
                 )
         raise KeyError(f"No such container found by container name: {self.container_name}")
+
+    def try_adopt_task_instances(self, tis: Sequence[TaskInstance]) -> Sequence[TaskInstance]:
+        """
+        Adopt task instances which have an external_executor_id (the ECS task ARN).
+
+        Anything that is not adopted will be cleared by the scheduler and becomes eligible for re-scheduling.
+        """
+        with Stats.timer("ecs_executor.adopt_task_instances.duration"):
```

Comment:
What kind of times are we seeing for this stat in your testing?

Reply:
Here's a quick demo.  The timestamps don't line up perfectly because of the delay in metric collection, but roughly:

| Time     | # Dags | Seconds to Adopt |
|----------|--------|------------------|
| 00:55:42 | 0      | 0.0000346        |
| 00:56:08 | 2      | 0.988            |
| 00:58:24 | 8      | 3.79             |
| 01:00:53 | 3      | 1.41             |
| 01:04:11 | 2      | 0.952            |
| 01:14:00 | 0      | 0.0000144        |

![image](https://github.com/apache/airflow/assets/1920178/2ac94a9a-7a5c-45e8-a5fd-2475dd868fd2)